qualtran/bloqs/qsp/generalized_qsp.py

```diff
@@ -274,16 +345,51 @@ def decompose_from_registers(
             yield self.U.adjoint().on_registers(**quregs)
 
     def build_call_graph(self, ssa: 'SympySymbolAllocator') -> Set['BloqCountT']:
-        degree = len(self.P)
+        degree = len(self.P) - 1
 
-        counts = {(rotation, 1) for rotation in self.signal_rotations}
+        counts = set(Counter(self.signal_rotations).items())
 
         if degree > self.negative_power:
-            counts.add((self.U.controlled(control_values=[0]), degree - self.negative_power))
+            counts.add((Controlled(self.U, CtrlSpec(cvs=0)), degree - self.negative_power))
```

Comment:
This should eventually be `self.U.controlled()` but the controlled infrastructure right now is in a messy state so it's fine to do this. Please open an issue to track this however.